analyzers/src/SonarAnalyzer.CSharp/Rules/XmlExternalEntityShouldNotBeParsed.cs

```diff
@@ -63,20 +65,20 @@ public sealed class XmlExternalEntityShouldNotBeParsed : SonarDiagnosticAnalyzer
                     ccc.RegisterSyntaxNodeActionInNonGenerated(
                         c =>
                         {
-                            var objectCreation = (ObjectCreationExpressionSyntax)c.Node;
+                            var objectCreation = ObjectCreationFactory.Create(c.Node);
                             var netFrameworkVersion = versionProvider.GetDotNetFrameworkVersion(c.Compilation);
                             var constructorIsSafe = ConstructorIsSafe(netFrameworkVersion);
 
                             var trackers = TrackerFactory.Create();
                             if (trackers.XmlDocumentTracker.ShouldBeReported(objectCreation, c.SemanticModel, constructorIsSafe)
                                || trackers.XmlTextReaderTracker.ShouldBeReported(objectCreation, c.SemanticModel, constructorIsSafe))
                             {
-                                c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, objectCreation.GetLocation()));
+                                c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, objectCreation.Expression.GetLocation()));
                             }
 
-                            VerifyXPathDocumentConstructor(c);
+                            VerifyXPathDocumentConstructor(c, objectCreation);
                         },
-                        SyntaxKind.ObjectCreationExpression);
+                        SyntaxKind.ObjectCreationExpression, SyntaxKindEx.ImplicitObjectCreationExpression);
```

Comment:
Shouldn't S2755 be part of this PR title as well? There's this new registration.